app/src/main/java/com/brandon/forzenbook/view/composables/CreateUserComposables.kt

```diff
@@ -46,104 +62,239 @@ fun CreateAccountContent(
+                BlackButton(text = resources.getString(R.string.createAccountCreateButtonText)) {
+                    keyboardController?.hide()
+                    if (onCheckConnection()) {
+                        if (firstName.isNotEmpty() && lastName.isNotEmpty() && dateOfBirth.isNotEmpty() && email.isNotEmpty() && location.isNotEmpty()) {
+                            onSubmit(firstName, lastName, dateOfBirth, email, location)
+                        } else emptyFieldError = true
+                    } else isDialogOpen = true
+                }
+                if (emptyFieldError) TextFieldErrorText(text = resources.getString(R.string.requiredFieldsError))
+            }
         }
-        InputInfoTextField(
-            hint = resources.getString(R.string.createAccountLocationHint),
-            onTextChange = { location = it },
-            characterLimit = LOCATION_CHAR_LIMIT,
-            onMaxCharacterLength = { locationError = it },
+    )
+    if (isDialogOpen) {
+        AlertDialog(
+            title = resources.getString(R.string.errorNoInternetConnection),
+            body = resources.getString(R.string.errorConnectAndTryAgain),
+            onDismissRequest = {
+                isDialogOpen = false
+            },
         )
-        if (state.isLocationError) {
-            TextFieldErrorText(text = resources.getString(R.string.createAccountLocationError))
-        }
-        BlackButton(
-            text = resources.getString(R.string.createAccountCreateButtonText),
-        ) {
-            keyboardController?.hide()
-            if (firstName.isNotEmpty() && lastName.isNotEmpty() && dateOfBirth.isNotEmpty() && email.isNotEmpty() && location.isNotEmpty()) {
-                onSubmit(firstName, lastName, dateOfBirth, email, location)
-            } else {
-                emptyFieldError = true
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateAccountLoading(
+    state: Loading = Loading(),
+) {
+    var firstName by rememberSaveable { mutableStateOf(state.firstName) }
+    var lastName by rememberSaveable { mutableStateOf(state.lastName) }
+    var dateOfBirth by rememberSaveable { mutableStateOf(state.dateOfBirth) }
+    var email by rememberSaveable { mutableStateOf(state.email) }
+    var location by rememberSaveable { mutableStateOf(state.location) }
+    val navController = LocalNavController.current
+    val resources = LocalContext.current.resources
+    val calendar = Calendar.getInstance()
+    val dialog = DatePickerDialog(
+        LocalContext.current,
+        R.style.MySpinnerStyle,
+        null,
+        calendar.get(Calendar.YEAR),
+        calendar.get(Calendar.MONTH),
+        calendar.get(Calendar.DAY_OF_MONTH)
+    )
+    dialog.setOnDateSetListener { _, year, month, dayOfMonth ->
+        dateOfBirth = "${month+1}/$dayOfMonth/$year"
+        dialog.hide()
+    }
+    Scaffold(
```

Comment:
Why is there so much in a "Loading" composable? What does loading mean here? It looks very similar to the other composable

Reply:
I have trimmed it down to make it more clear what this composable does. The loading composable is just a faster composable that is called while the viewModel passes along the info. 

As a high level overview, this really is just an empty, noninteractable view that shows info while the submission is processing.

Reply:
Trimmed some more fat out of these to make them slim, I had thought it would be faster to use simple composable that mirror the appearance of the Idle composable, since they need to just show information, and they don't persist on screen longer than while the operation is running.

Let me know if you feel like that is inappropriate for this or if I should take a different approach. I just figured that this would be a good balance between managing state cleaner and also reducing the amount of additional code needed.